tests/test_variance_plane.py

```diff
@@ -400,122 +510,23 @@ def buildExposure(
-                [exp_bbox.maxX - margin_x, exp_bbox.maxY - margin_y],
-                (len(source_params), 2),
-            ).tolist()
-
-            # Loop over the sources and draw them onto the image cutout by
-            # cutout.
-            for i, params in enumerate(source_params):
-                # Dynamically get constructor and remove type from params.
-                constructor = profile_constructors[params.pop("type")]
-
-                # Get shear parameters and remove them from params.
-                g1, g2 = params.pop("g1"), params.pop("g2")
-
-                # The extent of the cutout should be large enough to contain
-                # the entire object above the background level. Some empirical
-                # factor is used to mitigate artifacts.
-                half_extent = 10 * params["half_light_radius"] * (1 + 2 * np.sqrt(g1**2 + g2**2))
-
-                # Pass the remaining params to the constructor and apply shear.
-                galsim_object = constructor(**params).shear(galsim.Shear(g1=g1, g2=g2))
-
-                # Retrieve the position of the object.
-                x, y = self.positions[i]
-                pos = galsim.PositionD(x, y)
-
-                # Get the bounds of the sub-image based on the object position.
-                sub_image_bounds = galsim.BoundsI(
-                    *map(int, [x - half_extent, x + half_extent, y - half_extent, y + half_extent])
-                )
-
-                # Identify the overlap region, which could be partially outside
-                # the image bounds.
-                sub_image_bounds = sub_image_bounds & image.bounds
-
-                # Check that there is some overlap.
-                assert sub_image_bounds.isDefined(), "No overlap with image bounds"
-
-                # Get the sub-image cutout.
-                sub_image = image[sub_image_bounds]
-
-                # Draw the object onto the image within the the sub-image
-                # bounds.
-                galsim_object.drawImage(
-                    image=sub_image,
-                    offset=pos - sub_image.true_center,
-                    method="real_space",  # Save memory, usable w/o convolution
-                    add_to_image=True,  # Add flux to existing image
-                    scale=pixel_scale,
-                )
-
         # Add a constant background to the entire image (both in e-/pixel).
-        image += self.background
+        if add_signal:
+            image = self.raw_image + self.background
+        else:
+            image = self.signal_free_raw_image + self.background
 
         # Add noise to the image which is in electrons. Note that we won't
         # specify a `sky_level` here to avoid double-counting it, as it's
-        # already included as the background.
-        image.addNoise(galsim.PoissonNoise(rng))
+        # already included in the image as the background.
+        image.addNoise(galsim.PoissonNoise(self.rng))
```

Comment:
Is this meant to represent shot noise from which sources? Or read noise? (which should be Gaussian, although perhaps it doesn't matter that much). In any case, I thought Galsim had a CCD noise model (perhaps it doesn't matter that much for all this, I'm just curious).

Reply:
We decided to only consider the shot noise of the image for this idealized test, but I recall receiving a similar comment from @czwa regarding adding the read noise. This would probably be beyond the scope of this ticket, but perhaps in the future, we can make this test more realistic by adding the read noise as well, and GalSim has the capability to do so.